Client/Frontend/Login/LoginListViewController.swift

```diff
@@ -0,0 +1,352 @@
+    }
+    
+    private func fetchLoginInfo(_ searchQuery: String? = nil) {
+        guard !isFetchingLoginEntries else {
+            return
+        }
+        
+        isFetchingLoginEntries = true
+        
+        if let query = searchQuery {
+            profile.logins.getLoginsForQuery(query) >>== { [weak self] results in
+                self?.reloadEntries(results: results)
+            }
+        } else {
+            profile.logins.getAllLogins() >>== { [weak self] results in
+                self?.reloadEntries(results: results)
+            }
+        }
+    }
+    
+    private func reloadEntries(results: Cursor<Login>) {
+        loginEntries = results.asArray()
+        DispatchQueue.main.async {
+            self.tableView.reloadData()
+            self.isFetchingLoginEntries = false
+            self.navigationItem.rightBarButtonItem?.isEnabled = !self.loginEntries.isEmpty
+        }
+    }
+}
+
+// MARK: TableViewDataSource - TableViewDelegate
+
+extension LoginListViewController {
+    
+    override func numberOfSections(in tableView: UITableView) -> Int {
+        tableView.backgroundView = loginEntries.isEmpty ? emptyLoginView : nil
+
+        return Section.allCases.count
+    }
+    
+    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        if section == Section.options.rawValue {
+            return 1
+        } else {
+            return loginEntries.count
+        }
+    }
+    
+    override func tableView(_ tableView: UITableView, heightForRowAt indexPath: IndexPath) -> CGFloat {
+        if indexPath.section == Section.options.rawValue, searchController.isActive || tableView.isEditing {
+            return 0
+        }
+        return UITableView.automaticDimension
+    }
+    
+    override func tableView(_ tableView: UITableView, heightForHeaderInSection section: Int) -> CGFloat {
+        if section == Section.savedLogins.rawValue {
+            return loginEntries.isEmpty ? .zero : UX.headerHeight
+        }
+        
+        return .zero
+    }
+    
+    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        if indexPath.section == Section.options.rawValue {
+            let toggle = UISwitch().then {
+                $0.addTarget(self, action: #selector(didToggleSaveLogins), for: .valueChanged)
+                $0.isOn = Preferences.General.saveLogins.value
+            }
+            
+            let cell = tableView.dequeueReusableCell(withIdentifier: Constants.saveLoginsRowIdentifier, for: indexPath).then {
+                $0.textLabel?.text = Strings.saveLogins
+                $0.separatorInset = .zero
+                $0.accessoryView = toggle
+                $0.selectionStyle = .none
+            }
+            
+            return cell
+        } else {
+            let loginInfo = loginEntries[indexPath.item]
```

Comment:
access it with `safe: ` subscript

Reply:
Fix at https://github.com/brave/brave-ios/pull/4124/commits/ded5065d24f539a7dee6bb8bcb40b93a2b70fe7b